pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -715,17 +715,13 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 	}
 
 	if osdProps.onPVC() {
-		// the "all" placement is applied separately so it will have lower priority.
-		// We want placement from the storageClassDeviceSet to be applied and override
-		// the "all" placement if there are any overlapping placement settings.
-		c.spec.Placement.All().ApplyToPodSpec(&deployment.Spec.Template.Spec)
-		// apply storageClassDeviceSet Placement
-		// If nodeAffinity is specified both in the device set and "all" placement,
-		// they will be merged.
+		c.placementForOsdandPrepareOsdPods(&deployment.Spec.Template.Spec)
```

Comment:
The helper method could apply the other placement as well, right?
```suggestion
		c.placementForOsdandPrepareOsdPods(osdProps.placement, &deployment.Spec.Template.Spec)
```

Reply:
I don't think so, for PVCs-based osd it is of cephv1.Placement types and for non-PVCs, it is of cephv1.PlacementSpec